src/ring_buffer.zig

```diff
@@ -20,54 +23,74 @@ pub fn RingBuffer(comptime T: type, comptime size: usize) type {
         }
 
         /// Return but do not remove the next item, if any.
-        pub fn peek(self: *Self) ?T {
+        pub fn peek(self: *Self) ?*T {
             if (self.empty()) return null;
-            return self.buffer[self.index];
+            return &self.buffer[self.index];
         }
 
         /// Remove and return the next item, if any.
         pub fn pop(self: *Self) ?T {
             if (self.empty()) return null;
-            const ret = self.buffer[self.index];
-            self.index = (self.index + 1) % self.buffer.len;
-            self.count -= 1;
-            return ret;
+            defer {
+                self.index = (self.index + 1) % self.buffer.len;
+                self.count -= 1;
+            }
+            return self.buffer[self.index];
```

Comment:
This is sweet :)

Reply:
Cool :) Glad you agree. I was in two minds about it, since it's not resource deallocation and it's upside down... but we save coming up with a temp local scope variable name 😄